python/src/spark_rapids_ml/params.py

```diff
@@ -39,6 +39,28 @@
 P = TypeVar("P", bound="_CumlParams")
 
 
+class HasEnableSparseDataOptim(Params):
+
+    """
+    This is a Params based class inherited from XGBOOST: https://github.com/dmlc/xgboost/blob/master/python-package/xgboost/spark/params.py.
+    It holds the variable to store the boolean config of enabling sparse data optimization.
+    """
+
+    enable_sparse_data_optim = Param(
+        Params._dummy(),
+        "enable_sparse_data_optim",
+        "This stores the boolean config of enabling sparse data optimization, if enabled, "
+        "Spark rapids ml will construct a sparse matrix in the the format of csr_matrix, "
+        "then calls cuml with the sparse matrix. This config is disabled by default. If most of "
```

Comment:
Technically it is enabled by default as the default None value bases sparse vs dense off first row.

Reply:
Yeah, revised the docstring to be more accurate.